Xamarin.Forms.Core/DataTemplate.cs

```diff
@@ -49,10 +49,14 @@ public void SetValue(BindableProperty property, object value)
 			Values[property] = value;
 		}
 
-		internal override void SetupContent(object item)
+		internal override object OnCreateContent(object item, BindableObject container)
 		{
-			ApplyBindings(item);
-			ApplyValues(item);
+			var content = base.OnCreateContent(item, container);
+
+			ApplyBindings(content);
+			ApplyValues(content);
```

Comment:
I removed the internal `SetupContent` from the base and added it here. No need to have that method there just for this type.